linera-core/src/chain_worker/actor.rs

```diff
@@ -285,18 +288,21 @@ where
     )]
     pub async fn handle_requests(
         mut self,
-        mut incoming_requests: mpsc::UnboundedReceiver<ChainWorkerRequest<StorageClient::Context>>,
+        mut incoming_requests: mpsc::UnboundedReceiver<(
+            ChainWorkerRequest<StorageClient::Context>,
+            tracing::Span,
+        )>,
     ) {
         trace!("Starting `ChainWorkerActor`");
 
-        while let Some(request) = incoming_requests.recv().await {
+        while let Some((request, span)) = incoming_requests.recv().await {
             // TODO(#2237): Spawn concurrent tasks for read-only operations
             trace!("Handling `ChainWorkerRequest`: {request:?}");
 
             let responded = match request {
                 #[cfg(with_testing)]
                 ChainWorkerRequest::ReadCertificate { height, callback } => callback
-                    .send(self.worker.read_certificate(height).await)
+                    .send(self.worker.read_certificate(height).instrument(span).await)
```

Comment:
Should we deduplicate all the `instrument(span)`s (and also prevent that we forget adding it to calls in the future) by calling `handle_requests(…).instrument(span)` from `handle_requests_with_span`?

Reply:
I was on the fence about doing this anyway :sweat_smile: